clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java

```diff
@@ -0,0 +1,34 @@
+package io.lakefs;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class LakeFSFileStatus extends FileStatus {
+
+    private String checksum;
+    private String physicalAddress;
+
+    public LakeFSFileStatus(long length, boolean isdir, int block_replication,
+                            long blocksize, long modification_time, Path path,
+                            @Nullable String checksum, @Nullable String physicalAddress) {
+        super(length, isdir, block_replication, blocksize, modification_time, path);
+        if (!isdir) {
+            // directories are not objects in lakefs, therefore we only expect to have these details for non-directories
+            Objects.requireNonNull(checksum);
+            Objects.requireNonNull(physicalAddress);
```

Comment:
I don't understand why we need the `@Nullable` annotation _at all_, if we're going to check at runtime.  Maybe construct directories from some separate (probably named static) method?